### DISCUSSION TITLE
Quest "What is the structure of this bridge?" for single-span bridges

### DISCUSSION THREAD

**Discussion by deevroman** [streetcomplete/StreetComplete#5603] [General]
Today StreetComplete asked about the structure of this [bridge](https://www.openstreetmap.org/way/1004526875):

 <img src="https://upload.wikimedia.org/wikipedia/commons/d/d9/Sankt-Peterburg_avgust2013_1450051.JPG" width=50%> </img> 

 <img src="https://github.com/streetcomplete/StreetComplete/assets/25708359/4fe2d0e6-7be5-4241-bb93-26d2a7e53176" width="40%"> </img>

And I am quite unobvious that you need to choose the first option, because in the drawings all but two bridges have several spans.

Does anyone else stumble when solving this quest?

**Comment by matkoniecz** [comment]
I guess that single span graphic can be added?

**Reply by mnalis** [reply]
You mean, additional graphics which would be tagged the same? 

I think having two pictures that result in the same tagging would be much more confusing by itself then any original confusion (e.g. we've had similar issue in the past with e.g. "small" and "big" gate in barrier type quest in https://github.com/streetcomplete/StreetComplete/discussions/3729, https://github.com/streetcomplete/StreetComplete/issues/3717).

**Reply by matkoniecz** [reply]
> You mean, additional graphics which would be tagged the same?

yes

**Comment by mnalis** [comment]
I've always understood the question to ask about what kind of support is there that helps so the traveling surface does not collapse - i.e. the red parts in the pictures. 

And **not** about the number of pillars (or lack of them) that are under the bridge (it seems obvious to me that number will vary depending on the width of the river / water body that is being crossed). 

_(Then again, I was following that ticket when it was being implemented, so I'm hardly unbiased.)_

---

But perhaps the question text could be improved? Technically, _"[structure](https://en.wikipedia.org/wiki/Structure#Load-bearing)"_ would indeed imply _"load-bearing structure"_, and thus include pillars too (although few users are versed in structural engineering). 

Maybe adding _**"(ignore the number of pillars)"**_ to the question would make it clear? 
(then again, that might be somewhat long...)

**Reply by mnalis** [reply]
Or perhaps, if we need visual-only route, perhaps all the pillars in the pictures could be made with dashed lines for outline (instead of solid line)?

**Comment by kmpoppe** [comment]
The hint is in the wiki text for [`bridge:structure=beam`](https://wiki.openstreetmap.org/wiki/Tag:bridge:structure%3Dbeam):

"A bridge that has no additional support between both ends of entire bridge **or its section**. It **can** have pillars but **no additional support** between them."

Technically, the bridge shown by OP only has two [abutments](https://en.wikipedia.org/wiki/Abutment), at either end, where all the load is distributed to. 

The graphic used by SC (which is from [`bridge:structure=*`](https://wiki.openstreetmap.org/wiki/Key:bridge:structure)) is correct when you consider each part between the pillars as its own section where those sections are only transferring the load to the abutments either on land or on the pillar(s). It's still a beam bridge.

@mnalis's idea to dash the pillars in the middle might be the only way to communicate that these are optional, because I don't think it's possible to correctly map [`bridge:support=*`](https://wiki.openstreetmap.org/wiki/Key:bridge:support)-nodes in SC.
